.github/workflows/linting.yml

```diff
@@ -0,0 +1,63 @@
+name: Linting
+
+on:
+  pull_request:
+    branches:
+      - '**'
+
+concurrency:
+  group: linting-${{ github.sha }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  eslint:
+    name: ESLint
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        id: checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        id: setup-node
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: .node-version
+          cache: npm
+
+      - name: Install Dependencies
+        id: install
+        run: npm ci
+
+      - name: Lint
+        id: lint
+        run: |
+          npm exec eslint --max-warnings=0
+
+  editorconfig:
```

Comment:
Can you simplify this by merging these two jobs (`eslint` and `editorconfig`)? They both do pretty laborious things like installing Node (5-6s) and running `npm ci` (3-4s). The only difference is in the `Lint` steps, which take only 1s in both jobs. 

I propose doing only one job that - as before - installs Node, runs `npm ci`, and then has two `Lint` steps:

```yml
      - name: ESLint
        run: npm exec eslint --max-warnings=0

      - name: Lint .editorconfig
        run: npm exec ec -- -config .editorconfig-checker
```

This would run only 1s slower and would be easier to maintain (something we want to optimize for here).

Reply:
The reason they're two separate jobs is so that even if one job fails, the other can still report its success. If for example, I create a PR that fails both checks, I would fix the issues reported by the first check, only to find out that there were issues with the second. By separating each check into it's own job, we'd only have to fix issues caught by checks once. My team and I adopt this approach at work and it works pretty well for us, especially when we have 10+ checks that run in parallel.

We could create a shared action that encapsulates the "Checkout", "Setup Node", and "Install Dependencies" steps into a single action though! That would de-duplicate the boilerplate leading up to running each check.

Reply:
Yes, that's a pretty common problem! In my experience, if a CI job fails, I have to look at its CLI output anyways to see what exactly failed: which file, which line number, what's the exact error message, what else did go wrong. And when I fix something that broke on CI, I want to verify my fix locally instead of pushing to CI and waiting to see the results there. The latter is too slow.

I therefore usually make it super easy for developers to run the exact same tests locally that CI runs. Or -- even better -- have CI run the same test scripts that developers run. On Git Town there is one [Makefile](https://github.com/git-town/git-town/blob/main/Makefile) that both developers and CI call. For example, when CI runs unit tests, it calls [make unit-race](https://github.com/git-town/git-town/blob/28fa20a4e6f6821191d9b627946f3b50a36bbe33/.github/workflows/unit.yml#L17) the same way I as a developer run unit tests. It runs the same end-to-end tests that developers run because like developers it simply executes [make cuke](https://github.com/git-town/git-town/blob/28fa20a4e6f6821191d9b627946f3b50a36bbe33/.github/workflows/cuke.yml#L17-L18) to run them. When linting, CI also just runs [make fix and make docs](https://github.com/git-town/git-town/blob/28fa20a4e6f6821191d9b627946f3b50a36bbe33/.github/workflows/lint.yml#L17-L20) just as developers do.

So, when CI fails, I just run these tests/linters locally and I'll see the exact same errors and can verify their fixes.

My take is that both checks in your setup are so quick (1 sec each) that running both via a hypothetical `npm run lint` would not be much slower. But it would be much easier than looking up whether the `ESLint` or `Lint .editorconfig` job failed, and then running `npm exec eslint --max-warnings=0` or `npm exec ec -- -config .editorconfig-checker` locally.

Having said all this, I don't feel super strong about this. If you are used to this setup, then it's okay to leave it. Your setup is pretty nice as well and has its advantages.

Reply:
I see where you're coming from! I can see how it would be a lot easier if we made it easier for contributors to run CI checks locally. I'll take a note of it and come back around to it once I have a bit more time to make changes to the CI some more. Thanks for taking the time to explain your approach!

Reply:
Created issue to track: https://github.com/git-town/action/issues/2